src/main/java/dk/kb/datahandler/kaltura/KalturaDeltaUploadJob.java

```diff
@@ -0,0 +1,304 @@
+            solrQuery.set("spellcheck", false); //No spellcheck
+            solrQuery.add("sort", "internal_storage_mTime ASC"); // increasing order
+            solrQuery.add("fl", fieldList);
+            solrQuery.setRows(batchSize);
+            QueryResponse response = client.query(solrQuery);
+            SolrDocumentList results = response.getResults();
+            log.info("Load solr records for delta upload:" + results.getNumFound());
+            return results;
+        }
+    }
+
+    /**
+     * 
+     * 
+     * @param title  Title field for kaltura
+     * @param description Description field for kaltura
+     * @param fileId  This value will be used to calculate path of stream depending on originatesFrom
+     * @param originatesFrom      DOMS or Preservica
+     * @param resourceDescription VideoObject or AudioObjext
+     * @return The Kaltura entry id if upload is successful
+     * @throws IOException If upload fails
+     */
+    public static String uploadStream(String title, String referenceId, String description, String filePath, String tag, MediaType mediaType, int flavourParamId) throws IOException {
+
+        initKalturaClient();
+        log.info("Starting upload stream:" + filePath + " with flavorParamId:" + flavourParamId);
+        String entryId = kalturaClient.uploadMedia(filePath, referenceId, mediaType, title, description, tag, flavourParamId);
+        log.info("Upload completed for stream:" + filePath + " with filerefence:" + referenceId + " and got kaltura entryid:" + entryId);
+        return entryId;
+
+    }
+
+    /**
+     * Check if a file_id already does exist in kaltura. Then it is already uploaded.
+     * There can be meta-data errors where different records points to same stream.
+     * 
+     * @param file_id Our reference to the stream.
+     * @return kalturaId or null if does not exist.
+     * @throws IOException
+     * 
+     */
+    private static String getInternalIdKaltura(String file_id) throws IOException {
+
+        initKalturaClient();
+
+        String kalturaInternalId = kalturaClient.getKulturaInternalId(file_id);
+        if (kalturaInternalId != null) {
+            log.warn("Kaltura fileId:" + file_id + " is already in kaltura with entry_id:" + kalturaInternalId);
+
+        }
+        return kalturaInternalId;
+    }
+
+    /*
+     * Example: delta-2025-05-01 
+     * 
+     */
+     private static String getUploadTagForKaltura() {        
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd",Locale.getDefault());
+        String yyyyMMdd = formatter.format(new Date());
+        return "delta-"+yyyyMMdd;                
+    }
+    
+    /**
+     * Validate file exists on the file system and also minimum size in bytes.
+     * Will return null if everything is validated or error message on error.
+     * Errors are:
+     * <p>
+     * StreamErrorTypeDto.FILE_TOO_SHORT<b>
+     * StreamErrorTypeDto.FILE_MISSING
+     * @p
```

Comment:
Small typo. You probably mean </p>

Reply:
params, return added

Reply:
the param minimum in the javadoc should be minimumSizeInBytes as it is in the method